src/picongpu/include/particles/creation/creation.hpp

```diff
@@ -0,0 +1,81 @@
+/**
+ * Copyright 2015-2016 Heiko Burau
+ *
+ * This file is part of PIConGPU.
+ *
+ * PIConGPU is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PIConGPU is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PIConGPU.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "types.h"
+#include "particles/creation/creation.kernel"
+#include "simulation_defines.hpp"
+#include "cuSTL/algorithm/kernel/Foreach.hpp"
+#include "cuSTL/cursor/MultiIndexCursor.hpp"
+
+namespace picongpu
+{
+
+namespace particles
+{
+
+namespace creation
+{
+
+/** Calls the `createParticlesKernel` kernel to create new particles.
+ *
+ * @param sourceSpecies species from which new particles are created
+ * @param targetSpecies species of the new created particles
+ * @param particleCreator functor that defines the particle creation
+ * @param cellDesc mapping description
+ *
+ * `particleCreator` must define: `init()`, `numNewParticles()` and `operator()()`
+ * \see `PhotonCreator.hpp` for a further description.
+ */
+template<typename SourceSpecies, typename TargetSpecies, typename ParticleCreator,
+         typename CellDescription>
+void createParticles(SourceSpecies& sourceSpecies, TargetSpecies& targetSpecies, ParticleCreator particleCreator,
```

Comment:
Is `createParticles` the the right name for this function? Maybe it is something like `evaluateEachParticle` because the functor `ParticleCreator` can be an arbitrary operation.